### PR TITLE
dai-zephyr: do not set local buffer if no process func found

### PR DESCRIPTION
**Pull request by kv2019i**
Rework set_new_local_buffer() to not set dd->local_buffer multiple times.

### CODE REVIEW

**Comment by kv2019i** [comment]
Not sure @lyakh this helps much, but we do avoid one extra dd->local_buffer write.

**Review by juimonen — APPROVED** [review]
not sure I understand the optimization, but looks innocent enough to me

File: src/audio/dai-zephyr.c [hunk 1]
```diff
@@ -1468,32 +1468,33 @@ int dai_zephyr_multi_endpoint_copy(struct dai_data **dd, struct comp_dev *dev,
 static void set_new_local_buffer(struct dai_data *dd, struct comp_dev *dev)
 {
 	struct comp_buffer __sparse_cache *dma_buf = buffer_acquire(dd->dma_buffer);
+	struct comp_buffer __sparse_cache *local_buf_c;
 	struct comp_buffer __sparse_cache *local_buf;
```
Comment (lyakh) [inline src/audio/dai-zephyr.c:1472]: ouch, this one isn't `__sparse_cache`, there must be an error in sparse test.

**Review by lyakh — CHANGES_REQUESTED** [review]
looks good, thanks for fixing it! But need to fix one hick up.

**Comment by lyakh** [comment]
@kv2019i @juimonen Given two possibilities:
```
	never_fails();
	ret = might_fail();
	if (ret < 0) {
		undo_never_fails();
		return ret;
	}
```
and
```
	ret = might_fail();
	if (ret < 0)
		return ret;

	never_fails();
```
as long as the order indeed isn't important - often both will work correctly, but I think one is clearly preferable over the other and should be used whenever possible.

File: src/audio/dai-zephyr.c [hunk 2]
```diff
@@ -1468,32 +1468,33 @@ int dai_zephyr_multi_endpoint_copy(struct dai_data **dd, struct comp_dev *dev,
...
 
 	dd->process = pcm_get_conversion_function(local_fmt, dma_fmt);
 
-	if (!dd->process) {
+	if (dd->process)
+		dd->local_buffer = local_buf;
+	else
 		comp_err(dev, "converter function NULL: local fmt %d dma fmt %d\n",
 			 local_fmt, dma_fmt);
-		dd->local_buffer = NULL;
```
Comment (btian1) [inline src/audio/dai-zephyr.c:1495]: even line 1478, 1482 set this, line 1495 still can clear this setting, right?
seems this is a code refine, are you finding this with bug or code inspection?

File: src/audio/dai-zephyr.c (hunk 1 above)
Comment (kv2019i) [reply src/audio/dai-zephyr.c]: Fixed in V2.

**Review by lyakh — APPROVED** [review]

**Review by dnikodem — APPROVED** [review]

**Comment by kv2019i** [comment]
The Intel CI System/merge/build fails to one run of "TestSrcFullScope::test_01_04_src" indicating a glitch on one platform. I don't think it can be related to this PR, but need to do another run to be sure.

**Review by btian1 — APPROVED** [review]

**Review by tmleman — APPROVED** [review]

**Review by lgirdwood — APPROVED** [review]

**Comment by lgirdwood** [comment]
SOFCI TEST

**Comment by lgirdwood** [comment]
@kv2019i still failing CI, can you check. Thanks

**Pull request closed without merging**